llvm/lib/Target/AMDGPU/VOPInstructions.td

```diff
@@ -1960,6 +1957,16 @@ multiclass VOP3_Realtriple_t16_and_fake16_gfx12<bits<10> op, string asmName, str
   defm _fake16:VOP3_Realtriple_t16_gfx12<op, asmName, opName#"_fake16", pseudo_mnemonic, isSingle>;
 }
 
+multiclass VOP3Only_Realtriple_t16_gfx12<bits<10> op, string asmName,
+                                     string opName = NAME, string pseudo_mnemonic = "">
+  : VOP3_Realtriple_t16_gfx12<op, asmName, opName, pseudo_mnemonic, 1>;
+
+multiclass VOP3Only_Realtriple_t16_and_fake16_gfx12<bits<10> op, string asmName,
+                                     string opName = NAME, string pseudo_mnemonic = ""> {
+  defm _t16 : VOP3Only_Realtriple_t16_gfx12<op, asmName, opName#"_t16", pseudo_mnemonic>;
```

Comment:
Shouldn't `VOP3Only_Realtriple_t16/fake16_gfx12` do the `opName#"_t16/_fake16"` thing themselves?

Reply:
For now "t16_and_fake16" define t16 and fake16 together. 

In the end the fake16 will be removed and the change will be something like `VOP3Only_Realtriple_t16_and_fake16_gfx12` -> `VOP3Only_Realtriple_t16_gfx12`. There seems no necessary to add a fake16 function

Reply:
OK, I see.